site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Fields combining several inputs together with a fieldset, can be placed next to each other.</strong>
+    </p>
+    <div>For example:
+        <ul>
+            <li>Two to three radio buttons with short labels</li>
+            <li>First name / last name</li>
+            <li>Date (day/month/year)</li>
+            <li>Time (hour/minute/second)</li>
+        </ul>
+    </div>
+    <Image src="../../static/patterns/form/form-layout-horizontal@2x.png" alt="Fields combining several inputs together with a fieldset" style="max-width:468px;" viewable /> 
+</div>
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Use <a href="/design-tokens/spacing">spacing tokens</a> for grouping form controls with proximity and separation</strong>
+    </p>
+    <ul>
+        <li>Use smaller spacing between controls that belong together and larger spacing between groups of controls</li>
+    </ul>
+    <Image src="../../static/patterns/form/form-layout-spacing@2x.png" alt="Use spacing for grouping form controls" style="max-width:432px;" viewable /> 
+</div>
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Align form controls to <a href="/guidelines/grid">column grid</a>.</strong>
+    </p>
+    <Image src="../../static/patterns/form/form-layout-grid@2x.png" alt="Align fields to column grid" style="max-width:432px;" viewable /> 
+</div>
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Adjust field width according to preferred input.</strong>
+    </p>
+    <Image src="../../static/patterns/form/form-layout-width@2x.png" alt="Adjust field width according to preferred input" style="max-width:432px;" viewable /> 
+</div>
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <p>
+        <IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle', color: 'var(--color-success)' }} size="s"/>
+        <strong>Align form wide action buttons at the end of the form, aligned left in order of importance</strong>
```

Comment:
> Align form wide action buttons at the end of the form, aligned left in order of importance

Mikä mahtaa olla vallitseva käytäntö tällä hetkellä Helsingin palveluissa?

Reply:
Hmm… Tuskin on mitään käytäntöä. Aihanan voi olla tapauksia, joissa jotain secondary-actioneita tarvittais esim. lomakeen alussa. Tosin Tero tais erikseen todeta että tää on hyvä ohje.

Reply:
Muotoilin tän nyt vähän avoimemmaksi:
"**Group form wide action buttons together.** Best practice is to place action buttons at the end of the form, aligned to left and ordered by importance."